src/cliffordsynthesis/encoding/SATEncoder.cpp

```diff
@@ -50,22 +55,30 @@ void SATEncoder::createFormulation() {
   const auto start = std::chrono::high_resolution_clock::now();
   initializeSolver();
 
-  const std::size_t s = config.targetTableau->hasDestabilizers() &&
-                                config.initialTableau->hasDestabilizers()
-                            ? 2U * N
-                            : N;
+  S = config.targetTableau->hasDestabilizers() &&
+              config.initialTableau->hasDestabilizers()
+          ? 2U * N
+          : N;
 
-  tableauEncoder = std::make_shared<TableauEncoder>(N, s, T, lb);
+  tableauEncoder =
+      std::make_shared<TableauEncoder>(N, S, T, lb, config.ignoreRChanges);
   tableauEncoder->createTableauVariables();
   tableauEncoder->assertTableau(*config.initialTableau, 0U);
   tableauEncoder->assertTableau(*config.targetTableau, T);
 
+  if (config.ignoreRChanges) {
+    config.gateSet.removePaulis();
+    config.gateSet.emplace_back(qc::OpType::None);
```

Comment:
Multiple times throughout this PR I had the feeling that `None` is not consistently handled. If you have to add it back anyways, you could just not remove it in the first place? There is a lot of removing and adding `None` in the gate set handling that I believe could be consolidated and mostly eliminated.